packages/zowe-explorer/src/shared/refresh.ts

```diff
@@ -29,22 +26,23 @@ import { createJobsTree } from "../job/ZosJobsProvider";
  * @param {IZoweTree} treeProvider
  */
 export async function refreshAll(treeProvider: IZoweTree<IZoweTreeNode>) {
-    await readConfigFromDisk();
     await Profiles.getInstance().refresh(ZoweExplorerApiRegister.getInstance());
-    treeProvider.mSessionNodes.forEach(async (sessNode) => {
-        const setting = (await PersistentFilters.getDirectValue(
-            globals.SETTINGS_AUTOMATIC_PROFILE_VALIDATION
-        )) as boolean;
-        if (contextually.isSessionNotFav(sessNode)) {
-            labelRefresh(sessNode);
-            sessNode.children = [];
-            sessNode.dirty = true;
-            resetValidationSettings(sessNode, setting);
-            returnIconState(sessNode);
-            await syncSessionNode(Profiles.getInstance())((profileValue) =>
-                ZoweExplorerApiRegister.getCommonApi(profileValue).getSession()
-            )(sessNode);
-        }
-    });
-    treeProvider.refresh();
+    if (treeProvider.mSessionNodes.length > 0) {
+        treeProvider.mSessionNodes.forEach(async (sessNode) => {
```

Comment:
Why does it need to check the length? forEach will not execute the loop if array is empty.

Reply:
with the return from ProfilesCache arrays possibly being empty it is good to have a check for length of the sessionNodes in tree before the forEach call

Reply:
If treeProvider.mSessionNodes is an empty array, no need to check for length because forEach loop will not execute.

Reply:
addressed in commit https://github.com/zowe/vscode-extension-for-zowe/pull/1891/commits/42158f234ca77c0251a97aa22e2433b4f9c40b12